### PR TITLE
chore: Fix release_please_config script

### PR DESCRIPTION
**Pull request by lqiu96**
### Issues:
- The original script was including an extra comma in the last module -- Fixed the logic to generate valid JSON
- It also set an incorrect component value -- Fixed the GREP query to be more specific. i.e. 
 `"component": "proto-google-cloud-containeranalysis-v1",` -> `"component": "google-cloud-containeranalysis",`
This was also fixed with Deepankar's PR #8475 

### Changes
This is the pre-monorepo release flow (We plan on calling `./generation/update_versions.sh -s` one last time)
- Update the versions.txt for all the modules (./generation/update_versions.sh)
- Generate the release-please config files (./generation/generate_release_please_config.sh)
- Copy over the README.md and CHANGELOG.md (https://github.com/googleapis/google-cloud-java/pull/8498)

#### After the first release
When we have a new module in the monorepo, we can just call `./generation/generate_release_please_config.sh` to generate the `.json` files for release-please

Release-Please maintains the versions.txt and manifest.json files after every release. When we add in a new module, the module comes with a versions.txt. We can include the module and set the version in manifest.json. It'll be set to the released_version value in the module's versions.txt.

### CODE REVIEW

File: generation/generate_release_please_config.sh [hunk 1]
```diff
@@ -1,70 +1,60 @@
...
-  cat ${version_file} | while read line
-      do
-         if [[ ${line} =~ [0-9] ]] && ! [[ ${line} == *"SNAPSHOT"* ]] ; then
-           artifact_name=$(echo "${line}" | awk -F':' '{print $1}')
-           old_version=$(echo "${line}" | awk -F':' '{print $3}')
-           new_version=$(echo ${old_version} |  awk -F'.' '{print $1"."$2"."$3+1}' |  sed s/[.]$//)
-           new_version="${new_version}-SNAPSHOT"
-           sed -i.bak -e "s|${artifact_name}:${old_version}:${old_version}|${artifact_name}:${old_version}:${new_version}|" "${version_file}" && rm "${version_file}".bak
-         fi
-      done
```
Comment (suztomo) [inline generation/generate_release_please_config.sh:64]: Is this removed because we will use `./generation/update_versions.sh`? (Good)

**Review by suztomo — APPROVED** [review]

File: generation/generate_release_please_config.sh (hunk 1 above)
Comment (lqiu96) [reply generation/generate_release_please_config.sh:64]: Yep.

**Comment by lqiu96** [comment]
I'll keep this on hold while we investigate the monorepo release-please branch.

**Comment by suztomo** [comment]
Resolve .release-please-manifest.json

File: generation/generate_release_please_config.sh [hunk 2]
```diff
@@ -1,70 +1,60 @@
...
-rp_manifest_line=""
+echo "{" > "${release_please_manifest_file}"
 
-for module in $(find . -mindepth 2 -maxdepth 2 -name pom.xml |sort | xargs dirname); do
-
-  if [[ "${module}" = *google-cloud-gapic-bom ]] || [[ "${module}" = *CoverageAggregator ]]; then
+# We are manually excluding the CoverageAggregator module -- TODO: Update this for any other exclusions
+num_modules=$(find . -mindepth 2 -maxdepth 2 -name pom.xml | wc -l)
+num_modules=$((num_modules - 1))
+for path in $(find . -mindepth 2 -maxdepth 2 -name pom.xml | sort | xargs dirname); do
```
Comment (suztomo) [inline generation/generate_release_please_config.sh:20]: Can you add "--dictionary-order" parameter to `sort`?

https://github.com/googleapis/google-cloud-java/issues/8520

**Review by suztomo — CHANGES_REQUESTED** [review]
Fix sort parameter.

**Review by suztomo — APPROVED** [review]

**Comment by suztomo** [comment]
Retrying "Kokoro - Test: Integration" failure.

**Pull request merged**